modules/python/sngexample.py

```diff
@@ -45,12 +45,53 @@ def deinit(self):
     def send(self, msg):
         """Send a message to the target service
 
-        It should return True to indicate success, False will suspend the
-        destination for a period specified by the time-reopen() option."""
+        It can return either a boolean or integer.
+        Boolean: True to indicate success, False will suspend the
+        destination for a period specified by the time-reopen() option.
+        After that the same message is retried until retries() times.
+
+        Integer:
+        self.SUCCESS: message sending was successful (same as boolean True)
+        self.ERROR: message sending was unsuccessful. Same message is retried.
+            (same as boolean False)
+        self.DROP: message cannot be sent, it should be dropped immediately.
+        self.QUEUED: message is not sent immediately, it will be sent with the flush method.
+        self.NOT_CONNECTED: message is put back to the queue, open method will be called until success."""
+
```

Comment:
If the Java destination defines these values in the java side, i am wondering if we could do the same in Python.

That would simplify the python part and make this pretty explicit.

Reply:
Agreed. Injecting `template_options` was also a bit strange, but I solely for that i did not want to introduce inheritance through parent class. However, the problem is much more visible now, because of these newly injected return values, that are intended to be used by all user scripts.
After resolving https://github.com/balabit/syslog-ng/pull/2438, we should introduce a proper inheritance from a python destination parent class.